notebooks/circuits_with_bloqade.py

```diff
@@ -0,0 +1,763 @@
+    # One must first define the operator, then apply it to qubits.
+    # Directly building gates:
+    cx = squin.op.cx() # A controlled-X gate
+    cy = squin.op.cy() # A controlled-Y gate
+    cz = squin.op.cz() # A controlled-Z gate
+    ch = squin.op.ch() # A controlled-H gate
+    h = squin.op.h() # A Hadamard gate
+    s = squin.op.s() # An S gate
+    t = squin.op.t() # A T gate
+    X = squin.op.x() # The Pauli-X operator
+    Y = squin.op.y() # The Pauli-Y operator
+    Z = squin.op.z() # The Pauli-Z operator
+    sqrt_X = squin.op.sqrt_x() # The square root of the Pauli-X operator e^i pi/2 X
+    sqrt_Y = squin.op.sqrt_y() # The square root of the Pauli-Y operator e^i pi/2 Y
+    sqrt_Z = squin.op.sqrt_z() # The square root of the Pauli-Z operator e^i pi/2 Z
+    u = squin.op.u(theta=0.1, phi=0.2, lam=0.3) # An arbitrary single-qubit gate
+    
+    
+    # Non-unitary operators:
+    squin.op.p0() # Projector onto the |0> state
+    squin.op.p1() # Projector onto the |1> state
+    squin.op.spin_n() # The S- operator |0><1|
+    squin.op.spin_p() # The S+ operator |1><0|
+    squin.op.reset() # Reset a qubit to the |0> state
+    squin.op.phase(theta=0.1) # Add a global phase to the qubits
+    # Transforming gates into new operators:
+    squin.op.control(X,n_controls=2) # An X gate conditional on 2 qubits, aka a Toffoli gate
+    squin.op.rot(axis=X, angle=0.5) # A rotation around the X axis by 0.5 radians
+    squin.op.adjoint(t) # The inverse of a gate
+    
+    # Operators that are not unitaries to represent Paulis
+    squin.op.scale(op=X, factor=0.1) # Scale an operator by some factor, eg 0.1*X
+    squin.op.kron(lhs=X,rhs=Y) # The Kroneker product of two operators to act on multiple qubits, eg X ⊗ Y
+    squin.op.mult # Is this necessary?
+    squin.op.identity # The identity operator, which does nothing
+    squin.op.pauli_string(string="XXYZ") # A simpler alias for squin.op.kron(squin.op.X(), ...)
+
+
+# %% [markdown]
+# For example, lets make some operators and gates that go beyond the basic operators. Note that more complicated functionalities, such as T state teleportation, need function signatures that do not match that of the builtin `squin.gate` functions, and might need to be written using `op`. For example, consider a kernel that applies a controlled T gate to a register of qubits
+
+# %%
+try:
+    del controlled_t
+except:
+    pass
+try:
+    del heisenberg_rotation
+except:
+    pass
+@squin.kernel
+def controlled_t(qubits:Register):
+    """
+    A controlled T gate, aka a sqrt{CZ} gate.
+    ---o---
+       |
+    ---T---
+    """
+    t = squin.op.t()
+    ctrl = squin.op.control(t, n_controls=1)
+    squin.qubit.apply(operator=ctrl, qubits=qubits)
+
+controlled_t.print()
+
+# %% [markdown]
+# Finally, you can also define noise on qubits using the `squin.noise` namespace. Like `op`, `noise` operators are defined separately from their application to qubits
```

Comment:
FYI, stdlib noise short-hands are on my to-do list: https://github.com/QuEraComputing/bloqade-circuit/issues/451